### ISSUE TITLE
Illegal instruction error for dereplication command.

### ISSUE THREAD

**Issue by JHarrisonEcoEvo** [torognes/vsearch#341]
I am currently seeing an illegal instruction error when running the dereplication command:

vsearch --derep_fulllength some.fasta --output uniqueSequences.fa

I am using vsearch v2.9.0_linux_x86_64 with intel cpus (not exactly sure which model, I could look this up)

**Comment by mys721tx** [comment]
Could you try recompile VSEARCH on your machine? It seems to be a side-effect introduced by `-march=native` in the pre-compiled binary.

**Comment by JHarrisonEcoEvo** [comment]
cool will do

**Comment by colinbrislawn** [comment]
You could also try the vsearch version on conda.
```bash
conda install vsearch=2.9.0
```

**Comment by torognes** [comment]
Thank you for reporting this problem.

This is probably due to a change introduced with PR #327. I am going to reverse that change soon.

The distributed pre-compiled executables are unfortunately targeted to a specific machine that may have more features than the cpu you are running them on.

If you compile the software yourself it will be targeted to the machine were it is compiled. If you try to run it on another (older) machine you may see this error.

**Comment by torognes** [comment]
I have changed compiler options to `-march=x86-64` and `-mtune=generic` to produce code that will run on all x86_64 systems while tuning for more modern cpus. This is available in VSEARCH version 2.9.1 just released. Please download VSEARCH 2.9.1. It should fix this problem.

**Comment by frederic-mahe** [comment]
For dumb-dumbs like me, using `-march=x86-64` and `-mtune=generic` means that the compiler will build a code that can run on all computers with a `x86-64` architecture.

However, code that is optimal for a given micro-architecture can be slower for another. When facing a choice among different optimizations, the compiler will build code that favors the `generic` micro-architecture (i.e. the most common when the compiler was written). Again, that code will run on all computers with that architecture, but in a sub-optimal way for some micro-architectures.

Interestingly, what is `generic` will change depending on how old is the compiler. @torognes Should we then compile the release version of vsearch with a GCC version more recent than `GCC 4.4.7`?
```sh
objdump -s --section .comment vsearch 
```
```
vsearch:     format de fichier elf64-x86-64

Contenu de la section .comment :
 0000 4743433a 2028474e 55292034 2e342e37  GCC: (GNU) 4.4.7
 0010 20323031 32303331 33202852 65642048   20120313 (Red H
 0020 61742034 2e342e37 2d313829 00474343  at 4.4.7-18).GCC
 0030 3a202847 4e552920 342e392e 3200      : (GNU) 4.9.2.
```

**Comment by torognes** [comment]
The GCC version now used for compiling the released binaries is 4.9.2. It seems to be from October 2014 (https://gcc.gnu.org/releases.html). I have chosen to use this version because if I use a later version a newer set of libraries are required to run vsearch. I noticed that they were not available by default on the Linux system we are using, which is CentOS release 6.9. We have gcc version 4.9.2, 5.1.0, 5.2.0, 5.3.0, 6.1.0, 6.3.0, and 7.2.0 available (as well as some even older ones).

If I compile it with version 7.2.0 and try to run vsearch on CentOS 6.9 with only default libraries installed (without having loaded the gcc 7.2.0 module), I get the following errors:

```
-bash-4.1$ ./vsearch
./vsearch: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by ./vsearch)
./vsearch: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by ./vsearch)
```

It seems to require a version of libstdc++.so.6 supporting GLIBCXX_3.4.21, while the default library only supports GLIBCXX_3.4.13. It can easily be resolved by upgrading the libraries, but it could be a minor hassle for some users.

CentOS 6.9 is fairly old, but still supported.

I guess the speed improvement on new machines by going to a newer gcc version would be small, but I haven't tested it.

Do you think we should move on, or should we stay with 4.9.2 a bit longer?

**Comment by frederic-mahe** [comment]
All clusters I have access to seem to use CentOS, so gcc 4.9.2 sounds good to me.

CentOS 6.9 will be supported another two years (November 2020), we'll see what is the default gcc in CentOS 7 then. Meanwhile, we can continue to make sure vsearch compiles correctly with more recent gcc versions (next stop is `gcc 9.0` in 2019).

Advanced users who want to squeeze out a bit more performance can recompile vsearch using `-march=native`. By the way, is there an easy way to override the makefile parameters? Naively, I would try something like that:
```sh
./autogen.sh
./configure -march=native
make -j
```
